std/file.d

```diff
@@ -539,10 +539,20 @@ version(Posix) private void writeImpl(const(char)[] name, const(FSChar)* namez,
     cenforce(fd != -1, name, namez);
     {
         scope(failure) core.sys.posix.unistd.close(fd);
+
         immutable size = buffer.length;
-        cenforce(
-            core.sys.posix.unistd.write(fd, buffer.ptr, size) == size,
-            name, namez);
+        immutable chks = size_t.max / 2 + 1;
```

Comment:
Sorry, my original suggestion (`size_t.max/2+1`) was wrong. Should be `1<<30` (`uint.max/4+1`).